runtime/autoload/netrw.vim

```diff
@@ -365,7 +365,35 @@ call s:NetrwInit("g:netrw_cygdrive","/cygdrive")
 " Default values - d-g ---------- {{{3
 call s:NetrwInit("s:didstarstar",0)
 call s:NetrwInit("g:netrw_dirhistcnt"      , 0)
-call s:NetrwInit("g:netrw_decompress"       , '{ ".gz" : "gunzip", ".bz2" : "bunzip2", ".zip" : "unzip", ".tar" : "tar -xf", ".xz" : "unxz" }')
+call s:NetrwInit("g:netrw_decompress ", "{"
+            \ .."'.lz4':      'lz4 -d',"
+            \ .."'.lzo':      'lzop -d',"
+            \ .."'.lz':       'lzip -dk',"
+            \ .."'.7z':       '7za x',"
+            \ .."'.001':      '7za x',"
+            \ .."'.zip':      'unzip',"
+            \ .."'.bz':       'bunzip2 -k',"
+            \ .."'.bz2':      'bunzip2 -k',"
+            \ .."'.gz':       'gunzip -k',"
+            \ .."'.lzma':     'unlzma -T0 -k',"
+            \ .."'.xz':       'unxz -T0 -k',"
+            \ .."'.zst':      'zstd -T0 -d',"
+            \ .."'.Z':        'uncompress -k',"
+            \ .."'.tar':      'tar -xvf',"
+            \ .."'.tar.bz':   'tar -xvjf',"
+            \ .."'.tar.bz2':  'tar -xvjf',"
+            \ .."'.tbz':      'tar -xvjf',"
+            \ .."'.tbz2':     'tar -xvjf',"
+            \ .."'.tar.gz':   'tar -xvzf',"
+            \ .."'.tgz':      'tar -xvzf',"
+            \ .."'.tar.lzma': '"..(has('unix')?"XZ_OPT=-T0":(has("win32") && &shell =~? '\vcmd(\.exe)?$' ?"setx XZ_OPT=-T0 &&":"")).." tar -xvf --lzma',"
```

Comment:
Can you do the check for unix or windows once and than concatenate it? Also a bit of spacing in the concatenate would be nice... It is really hard to read.